pkg/pillar/cmd/zedrouter/patchenvelopes_test.go

```diff
@@ -75,12 +75,29 @@ func TestPatchEnvelopes(t *testing.T) {
 		}
 	}()
 
-	peStore.Wg.Add(1)
 	go func() {
 		peStore.PatchEnvelopeInfoCh <- peInfo
 	}()
 
-	peStore.Wg.Wait()
+	finishedProcessing := make(chan struct{})
+	go func() {
+		for {
+			if len(peStore.Get(u).Envelopes) > 0 && len(peStore.Get(u).Envelopes[0].BinaryBlobs) >= 2 {
+				close(finishedProcessing)
+				return
+			}
+			time.Sleep(time.Second)
+		}
+	}()
+
+	g.Eventually(func() bool {
+		select {
+		case <-finishedProcessing:
+			return true
+		case <-time.After(time.Minute):
+			return false
+		}
+	}, 2*time.Minute, time.Second).Should(gomega.BeTrue())
```

Comment:
What happens after 2 minutes? I guess this is just a unit test but I don't know what is the normal time it takes hence whether this can cause spurious failures due to load on the runner.

Reply:
after 2 minutes test will fail. Locally it takes around 1.016-1.018 for this test to pass.

Reply:
ok